### ISSUE TITLE
adaptation for Hebrew

### ISSUE THREAD

**Issue by dstoekl** [raphael-baena/morphology4metrology#2]
Hi there. Tx a lot for providing the pipeline whose results look very interesting also for Hebrew. I am trying to adapt this for Hebrew. I / Claude have created 
1. an alto loader that can either load the data directly for training or produce json+imagefolder pairs.
2. added a switch to enable RTL scripts. 
3. created visualization scripts 

My / Claude's patches and new code is here: https://github.com/dstoekl/hebrew-letter-spotting

4. After some unsuccesful attempts to pretrain and finetune on your Latin basemodel, I trained a new dtlr foundation model on a very limited codec {27 hebrew letters plus space, slash, dot and colon} based on 50k artificial lines of hebrew print, which seems to have worked. 
e.g. json: {"labels": [13, 29, 4, 11, 7, 8, 4, 28, 25, 13, 30, 15, 19, 11, 26, 9, 30, 16, 9, 9, 18, 18, 16, 22, 26, 18, 9, 29, 16, 4, 16, 24, 20, 13, 29, 20, 5, 28, 4, 15, 28, 30, 9, 5, 6, 13, 12, 13, 19, 16, 4], "boxes": [[1153.31, 35, 1168.31, 57], [1127.19, 34, 1153.19, 57], [1092.61, 35, 1118.61, 58], [1066.48, 34, 1092.48, 58], [1046.06, 34, 1066.06, 57], [1011.66, 34, 1037.66, 58], [985.48, 35, 1011.48, 58], [965.06, 34, 985.06, 57], [944.58, 35, 964.58, 65], [921.48, 35, 936.48, 57], [895.86, 34, 921.86, 57], [875.44, 35, 895.44, 57], [860.75, 35, 875.75, 66], [826.22, 34, 852.22, 58], [805.88, 35, 825.88, 57], [791.23, 35, 806.23, 58], [765.61, 34, 791.61, 57], [736.78, 29, 756.78, 57], [722.14, 35, 737.14, 58], [699.09, 35, 714.09, 58], [672.97, 35, 698.97, 58], [646.84, 35, 672.84, 58], [618.02, 29, 638.02, 57], [597.59, 35, 617.59, 60], [577.25, 35, 597.25, 57], [551.12, 35, 577.12, 58], [536.48, 35, 551.48, 58], [501.95, 34, 527.95, 57], [481.53, 29, 501.53, 57], [455.36, 35, 481.36, 58], [426.53, 29, 446.53, 57], [405.98, 35, 426.98, 57], [391.3, 35, 406.3, 57], [376.61, 35, 391.61, 57], [342.08, 34, 368.08, 57], [327.39, 35, 342.39, 57], [307.11, 35, 327.11, 57], [286.69, 34, 306.69, 57], [260.52, 35, 286.52, 58], [231.69, 35, 251.69, 57], [211.27, 34, 231.27, 57], [185.64, 34, 211.64, 57], [171.0, 35, 186.0, 58], [142.31, 35, 162.31, 57], [127.62, 34, 142.62, 58], [112.94, 35, 127.94, 57], [86.81, 34, 112.81, 57], [72.12, 35, 87.12, 57], [57.44, 35, 72.44, 66], [28.61, 29, 48.61, 57], [2.44, 35, 28.44, 58]], "orig_size": [73, 1186], "idx": 33514}
or a visualisation of another img:

<img width="526" height="69" alt="Image" src="https://github.com/user-attachments/assets/c489f5db-5436-4f45-8ac9-8815bfb211c3" />

5. I have tried to pretrain on it with manuscript images using my alto pipeline, but all results for 20 epochs remained noisy, too. I tried several times, once on top of the DTLR checkpoint after epoch 1 and once on top of the DTLR checkpoint after epoch 3. I continued training the DTLR model. 

6. I have now trained a new DTLR foundation model saving more epochs. If I apply it to print from the val set or to two well written manuscripts, it finds some characters in the manuscripts, but by far not all of them. Can I pretrain the m4m model on this or should I have started from a different procedure only inside the m4m repo?
